dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/event/AbstractEventService.java

```diff
@@ -2048,12 +2066,12 @@ private String getValidUsername( String userName, ImportSummary importSummary, S
         {
             validUsername = User.getSafeUsername( fallbackUsername );
         }
-        else if ( validUsername.length() >= 31 )
+        else if ( validUsername.length() > 255 )
```

Comment:
Can we move 255 to a constant e.g. on the UserCredentials class, e.g. USERNAME_MAX_LENGTH ?